plutus-ledger-api/src/PlutusLedgerApi/V1/Value.hs

```diff
@@ -368,6 +364,118 @@ split (Value mp) = (negate (Value neg), Value pos) where
   splitIntl mp' = These l r where
     (l, r) = Map.mapThese (\i -> if i <= 0 then This i else That i) mp'
 
+-- | The type of values that 'matchKVs' returns. See its Haddock for details.
+data MatchResult k v
+    = MatchSuccess
+    | MatchPartial [(v, v)] [(k, v)] [(k, v)]
+    | MatchFailure
+
+{-# INLINE matchKVs #-}
+{- | Take a function checking whether a value is zero\/empty, a function checking /structural/
```

Comment:
```suggestion
{- | Takes
```

Reply:
I like the first sentence to have imperative mood. I don't mind changing it, but I do that for virtually every Haddock that I wrote, so if I do it only here, it would only make things inconsistent.

I don't mind changing this habit, if it's something that we want. @zliu41 @michaelpj do you have an opinion?